src/data/bool/basic.lean

```diff
@@ -167,8 +167,6 @@ instance : linear_order bool :=
   le_antisymm := dec_trivial,
   le_total := dec_trivial,
   decidable_le := infer_instance,
-  decidable_eq := infer_instance,
-  decidable_lt := infer_instance,
```

Comment:
This doesn't look right to me.

Reply:
This is related to https://github.com/leanprover-community/lean/commit/47aeedd9472197a461139a482577f4ff837396a6. `linear_order` provides default implementation for those, implemented in terms of `decidable_le`: https://github.com/leanprover-community/lean/blob/master/library/init/algebra/order.lean#L203

Reply:
To be clear, I removed this because the `decidable_lt := infer_instance` line now errors out:
```
exact tactic failed, type mismatch, given expression has type
   true
 but is expected to have type
   as_true (∀ {x y : bool}, x < y ↔ x = ff ∧ y = tt)
 state:
 ⊢ as_true (∀ {x y : bool}, x < y ↔ x = ff ∧ y = tt) (lean-checker)
 ```

Reply:
The first `infer_instance` should remain, it's finding a "better" instance. You're right that the second one is now unnecessary

Reply:
Many thanks for the explanation! I added the first `infer_instance` back. I couldn't find a short way to make the second one work explicitly, since it complains about not being able to synthesize a preorder instance. Suggestions are welcome!